crates/core/tedge/src/bridge/common_mosquitto_config.rs

```diff
@@ -137,6 +137,7 @@ impl CommonMosquittoConfig {
 
         self.internal_listener.write(writer).await?;
         self.external_listener.write(writer).await?;
+        writer.flush().await?;
```

Comment:
note: if the caller calls `serialize` multiple times, then adding `flush` here means some of the flushes will be redundant. The point is to flush the writer before it's dropped, so the caller should probably do it either way, but keeping it here is also fine.